### PR TITLE
Add bytes.sent metric to count bytes sent between helpers

### PR DESCRIPTION
**Pull request by akoshelev**
A companion to `records.sent` that measures the total communication in bytes between helpers. With recent developments in IPA, the total number of records may stay the same, but size of messages sent is reduced. `records.sent` does not capture it, so we would rely on `bytes.sent` to check those improvements.

### CODE REVIEW

File: src/telemetry/step_stats.rs [hunk 1]
```diff
@@ -35,13 +35,13 @@ impl CsvExporter for Metrics {
         // then dump them to the provided Write interface
         // TODO: include role dimension. That requires rethinking `Metrics` implementation
         // because it does not allow such breakdown atm.
-        writeln!(w, "Step,Records Sent,Indexed PRSS,Sequential PRSS")?;
+        writeln!(w, "Step,Bytes sent,Indexed PRSS,Sequential PRSS")?;
```
Comment (benjaminsavage) [inline src/telemetry/step_stats.rs:38]: Why not both?

File: src/protocol/ipa/mod.rs [hunk 2]
```diff
@@ -860,19 +864,25 @@ pub mod tests {
                 .reconstruct();
 
             let snapshot = world.metrics_snapshot();
-            let records_sent = snapshot.get_counter(RECORDS_SENT);
-            let semi_honest_baseline = if per_user_cap == 1 {
-                RECORDS_SENT_SEMI_HONEST_BASELINE_CAP_1
+            let (records_baseline, bytes_baseline) = if per_user_cap == 1 {
+                (
+                    RECORDS_SENT_SEMI_HONEST_BASELINE_CAP_1,
+                    FIELD_SIZE * RECORDS_SENT_SEMI_HONEST_BASELINE_CAP_1,
```
Comment (benjaminsavage) [inline src/protocol/ipa/mod.rs:870]: I don’t think this is right. Currently IPA runs two separate validators, each over a subset of the protocol, so some records are sent in field 1 and others in field 2

File: src/telemetry/step_stats.rs (hunk 1 above)
Comment (akoshelev) [reply src/telemetry/step_stats.rs:38]: I was thinking that number of records may no longer be relevant now, but I don't mind adding it back

Comment (benjaminsavage) [reply src/telemetry/step_stats.rs:38]: I think it’s really helpful for analyzing where we should focus on optimizations.

File: src/protocol/ipa/mod.rs (hunk 2 above)
Comment (akoshelev) [reply src/protocol/ipa/mod.rs]: yep, my rebase was 4 hours old and stale already!

**Review by benjaminsavage — APPROVED** [review]
Thanks Alex, this is super useful. I'm already building all my new PRs on top of it, so I'm happy to merge it!

**Pull request merged**